src/node_report_module.cc

```diff
@@ -0,0 +1,295 @@
+  Local<Value> events_unchecked;
+  if (!obj->Get(context, eventskey).ToLocal(&events_unchecked)) return;
+  Local<Array> events;
+  if (events_unchecked->IsUndefined() || events_unchecked->IsNull()) {
+    events_unchecked = Array::New(env->isolate(), 0);
+    if (obj->Set(context, eventskey, events_unchecked).IsNothing()) return;
+  }
+  events = events_unchecked.As<Array>();
+
+  // Signal
+  Local<String> signalkey = env->signal_string();
+  Local<Value> signal_unchecked;
+  if (!obj->Get(context, signalkey).ToLocal(&signal_unchecked)) return;
+  Local<String> signal;
+  if (signal_unchecked->IsUndefined() || signal_unchecked->IsNull())
+    signal_unchecked = signalkey;
+  signal = signal_unchecked.As<String>();
+
+  Utf8Value signalstr(env->isolate(), signal);
+
+  // Report file
+  Local<String> filekey = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
+  Local<Value> file_unchecked;
+  if (!obj->Get(context, filekey).ToLocal(&file_unchecked)) return;
+  Local<String> file;
+  if (file_unchecked->IsUndefined() || file_unchecked->IsNull())
+    file_unchecked = filekey;
+  file = file_unchecked.As<String>();
+
+  Utf8Value filestr(env->isolate(), file);
+
+  // Report file path
+  Local<String> pathkey = FIXED_ONE_BYTE_STRING(env->isolate(), "path");
+  Local<Value> path_unchecked;
+  if (!obj->Get(context, pathkey).ToLocal(&path_unchecked)) return;
+  Local<String> path;
+  if (path_unchecked->IsUndefined() || path_unchecked->IsNull())
+    path_unchecked = pathkey;
+  path = path_unchecked.As<String>();
+
+  Utf8Value pathstr(env->isolate(), path);
+
+  // Report verbosity
+  Local<String> verbosekey = FIXED_ONE_BYTE_STRING(env->isolate(), "verbose");
+  Local<Value> verbose_unchecked;
+  if (!obj->Get(context, verbosekey).ToLocal(&verbose_unchecked)) return;
+  Local<Boolean> verbose;
+  if (verbose_unchecked->IsUndefined() || verbose_unchecked->IsNull())
+    verbose_unchecked = Boolean::New(env->isolate(), "verbose");
+  verbose = verbose_unchecked.As<Boolean>();
+
+  bool verb = verbose->BooleanValue(context).FromJust();
+
+  if (sync) {
+    static const std::string e = "exception";
+    static const std::string s = "signal";
+    static const std::string f = "fatalerror";
+    for (uint32_t i = 0; i < events->Length(); i++) {
+      Local<Value> v;
+      if (!events->Get(context, i).ToLocal(&v)) return;
+      Local<String> elem;
+      if (!v->ToString(context).ToLocal(&elem)) return;
+      String::Utf8Value buf(env->isolate(), elem);
+      if (*buf == e) {
+        options->report_uncaught_exception = true;
+      } else if (*buf == s) {
+        options->report_on_signal = true;
+      } else if (*buf == f) {
+        options->report_on_fatalerror = true;
+      }
+    }
+    if (*signalstr != nullptr) options->report_signal = *signalstr;
+    if (*filestr != nullptr) options->report_filename = *filestr;
+    if (*pathstr != nullptr) options->report_directory = *pathstr;
```

Comment:
again, the previous comment has no `reply` box!

> It would be good to split these two things;

will do.

>> not for debugging context

> But … this is a debugging utility?

Sorry; did not follow this statement - where is this quoted? I cannot spot this line anywhere.

Reply:
> How confident are you that this works the way you want it to?

Do you mean to state that these pointers point to V8 allocated strings and can never be null under normal circumstances?

Reply:
and to say that we don't want the nullchecks?

Reply:
> again, the previous comment has no `reply` box!

That usually happens when it’s already a reply to another review comment

> >> not for debugging context

> > But … this is a debugging utility?

> Sorry; did not follow this statement - where is this quoted? I cannot spot this line anywhere.

It’s from the previous comment: https://github.com/nodejs/node/pull/22712#discussion_r231948189

> > How confident are you that this works the way you want it to?

> Do you mean to state that these pointers point to V8 allocated strings and can never be null under normal circumstances?

I think the only reason for this to be nullptr is an OOM situation, but in that case we probably don’t want to silently ignore the errors? What is it that you did want to do with these checks?

Reply:
ok, so either:
 - leave the nullcheck and let the OOM take its natural route to crash
or
- introduce CHECK_NOT_NULL on the pointers.

I am leaning towards the CHECK_NOT_NULL, but LMK.

Reply:
Yes, `CHECK_NOT_NULL` should at least be fine, I think.

Reply:
this is done now, thnx.